src/org/labkey/targetedms/parser/blib/BlibSpectrumReader.java

```diff
@@ -510,6 +517,67 @@ private static BlibSpectrum readRedundantSpectrum(Connection conn, int redundant
         }
     }
 
+    // Return a map of (blib file) -> (spectrum source files, id files)
+    public static Map<String, BlibSourceFiles> readBlibSourceFiles(ITargetedMSRun run)
+    {
+        Map<PeptideSettings.SpectrumLibrary, Path> libs = LibraryManager.getLibraryFilePaths(run.getId());
+
+        Map<String, BlibSourceFiles> m = new TreeMap<>();
+        for (Map.Entry<PeptideSettings.SpectrumLibrary, Path> entry : libs.entrySet())
+        {
+            if (!entry.getKey().getLibraryType().contains("bibliospec_lite"))
+                continue;
+
+            Path path = entry.getValue();
+            String blibFilePath = getLocalBlibPath(run.getContainer(), path);
+            if (null == blibFilePath)
+                continue;
+
+            Set<String> sourceFiles = new HashSet<>();
+            Set<String> idFiles = new HashSet<>();
+            try (Connection conn = getBlibConnection(blibFilePath))
+            {
+                if (!hasTable(conn, "SpectrumSourceFiles"))
+                {
+                    continue;
+                }
+                try (Statement stmt = conn.createStatement(); ResultSet rs = stmt.executeQuery("SELECT * FROM SpectrumSourceFiles"))
+                {
+                    int fileNameColumn = -1;
+                    int idFileNameColumn = -1;
+                    ResultSetMetaData metadata = rs.getMetaData();
+                    for (int i = 1; i <= metadata.getColumnCount(); i++)
+                    {
+                        switch (metadata.getColumnName(i).toLowerCase())
+                        {
+                            case "filename":
+                                fileNameColumn = i;
+                                break;
+                            case "idfilename":
+                                idFileNameColumn = i;
+                                break;
+                        }
+                    }
+                    while (rs.next())
+                    {
+                        String fileName = rs.getString(fileNameColumn);
+                        String idFileName = idFileNameColumn >= 0 ? rs.getString(idFileNameColumn) : null;
+                        if (!StringUtils.isBlank(fileName) && !fileName.equalsIgnoreCase(idFileName))
```

Comment:
Maybe leave a comment here explaining fileName.equalsIngoreCase(idFileName) check?  You might also consider deferring this check to the panoramapublic module.